pkg/topology/run.go

```diff
@@ -537,6 +539,44 @@ func (tj *TopologyJob) Run(job job.JobRuntime) error {
 
 	query.FlushTopologyCache()
 
+	if t.Spec.PushLocation.URL != "" {
+		for _, c := range results {
+			if err := pushTopologyToLocation(job.Context, t.Spec.PushLocation, *c); err != nil {
+				job.History.AddErrorf("error pushing topology: %v", err)
+			}
+		}
+	}
+
+	return nil
+}
+
+func pushTopologyToLocation(ctx dutyCtx.Context, target v1.PushLocation, comp pkg.Component) error {
+	client := http.NewClient()
```

Comment:
I don't think this caters for oauth / bearer / or tls settings - we should probably extract checks/http.go:generateHTTPRequest into createHttpClient(types.Authentication) or types.Authentication.createCHttpClient()